src/main/webapp/scripts/map.js

```diff
@@ -187,20 +187,19 @@ function placeMarkerAndPanTo(latLng, map) {
 }
 
 /* Function to place markers from the datastore */
-function placeMarker(latLng, map) {
+function placeMarker(markerDetails, map) {
   const marker = new google.maps.Marker ({
-    position: latLng,
+    position: new google.maps.LatLng(markerDetails.lat, markerDetails.lng),
     map: map,
     icon: greyIcon, // TODO change the colour of the icon depending on the category
   });
+  marker.set("id", markerDetails.id);
 }
 
 /* Set the PopUp to Active */
-function togglePopup(latLng) {
+function togglePopup() {
```

Comment:
how do we pre-fill the "location" input ? is it somewhere else, not every toggling the popup?

Reply:
it's in the populateLocation function, called within the placeMarkerAndPanTo function - this hasn't changed in many CLs.

Reply:
right, thank you for the clarification :)